### PR TITLE
Implementation of Testing Framework

### PR DESCRIPTION
**Pull request by ChaseRun**
This PR converts all of the files in the examples directory into go test files. This was done so that in the future, we can have robust unit tests. Instead of "hardcoding" testing variables into each test function, they are now located
in a .env folder. The testing structure was inspired by the Terraform-Provider-IBM repository.

I also removed some unused packages / files.

Changes
 - Created unit tests in "test" directory
    - Deleted examples directory
    - Added ".vscode" directory for test configurations and env variables
    - Added private .env files to .gitignore
 - Removed helpers/env.go. Functions were not being used:
    - EnvFallBack()
 - Removed utils package. Functions were longer used:
    - GetEndpoint()
    - getGCEndpoint()
    - getNGEndpoint()
    - GetNext()
    - Relocated GetPowerEndPoint() from utils to ibmpisession
- Moved logger_test.go to new "test" directory
    - Renamed as ibmpi_logger_test.go
    - Made SanatizeArgs() in ibmpisession an exported function
- Updated vendor directory
- Updated version.go
- Updated go-sdk-core in go.mod to v5.8.2
- Edited README.md

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (yussufsh) [inline .vscode/launch.json:1]: We don't commit such workspace files. It is at user discretion to use what method he may like for testing.

If you really want this to be published then I would say write a document/wiki for how to run tests via vscode and example env files.

File: go.mod [hunk 2]
```diff
@@ -3,11 +3,17 @@ module github.com/IBM-Cloud/power-go-client
 go 1.13
 
 require (
-	github.com/IBM/go-sdk-core/v5 v5.8.0
+	github.com/IBM/go-sdk-core/v5 v5.9.1
 	github.com/apparentlymart/go-cidr v1.1.0
+	github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d // indirect
```
Comment (yussufsh) [inline go.mod:8]: Did you run `go mod tidy`?

File: ibmpisession/ibmpowersession.go [hunk 3]
```diff
@@ -80,34 +78,30 @@ func powerJSONConsumer() runtime.Consumer {
 	})
 }
 
-// Create a IBMPISession
-func NewSession(options *PIOptions) (*IBMPISession, error) {
+// Create a IBMPISession using an API Key
```
Comment (yussufsh) [inline ibmpisession/ibmpowersession.go:81]: NewSessionApiKey is not specific to API key only. We can also pass BearerToken Authenticator via options here. Hence name change is not needed here.

File: ibmpisession/logger.go [hunk 4]
```diff
@@ -24,10 +24,10 @@ func (IBMPILogger) Debugf(format string, args ...interface{}) {
 	if len(format) == 0 || format[len(format)-1] != '\n' {
 		format += "\n"
 	}
-	fmt.Fprintf(os.Stderr, format, sanatizeArgs(args)...)
+	fmt.Fprintf(os.Stderr, format, SanatizeArgs(args)...)
```
Comment (yussufsh) [inline ibmpisession/logger.go:27]: Want is the need here?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: ibmpisession/ibmpowersession.go [hunk 5]
```diff
@@ -39,8 +37,12 @@ type IBMPISession struct {
 
 // PIOptions
 type PIOptions struct {
+	// The authenticator used to configure the appropriate type of authentication
+	// Required
+	Authenticator core.Authenticator
+
 	// Enable/Disable http transport debugging log
-	Debug bool
+	Debug bool `default:true`
```
Comment (mkumatag) [inline ibmpisession/ibmpowersession.go:45]: This shouldn't be set by deafult

File: ibmpisession/logger.go [hunk 6]
```diff
@@ -24,10 +24,10 @@ func (IBMPILogger) Debugf(format string, args ...interface{}) {
...
 }
 
-func sanatizeArgs(args []interface{}) (out []interface{}) {
+func SanatizeArgs(args []interface{}) (out []interface{}) {
```
Comment (mkumatag) [inline ibmpisession/logger.go:30]: Why are making this public?

File: test/ibmpi_logger_test.go [hunk 7]
```diff
@@ -1,11 +1,13 @@
-package ibmpisession
```
Comment (mkumatag) [inline test/ibmpi_logger_test.go:1]: Unit tests are meant to stay with the source package, let's move to them into a different package

**Review by mkumatag — CHANGES_REQUESTED** [review]
It will be good we we can categories tests as unit and e2e and invoke them as seperately..

File: ibmpisession/ibmpowersession.go (hunk 3 above)
Comment (ChaseRun) [reply ibmpisession/ibmpowersession.go:81]: Fixed

File: ibmpisession/ibmpowersession.go (hunk 5 above)
Comment (ChaseRun) [reply ibmpisession/ibmpowersession.go:45]: Fixed

File: go.mod (hunk 2 above)
Comment (ChaseRun) [reply go.mod:8]: Running `go mod tidy` did not update this at all.

File: ibmpisession/logger.go (hunk 4 above)
Comment (ChaseRun) [reply ibmpisession/logger.go:27]: I made this uppercase so it can be an exported function and it can be used in the testing directory.

**Pull request closed without merging**